apps/dashboard/src/utils/liquid-scope-analyzer.ts

```diff
@@ -49,26 +75,37 @@ export function isVariableInLocalContext(content: string, name: string, position
   // Get all variables available at this position
   const availableVariables = getVariablesAtPosition(content, position);
 
-  return availableVariables.has(rootName);
+  return availableVariables.includes(rootName) || availableVariables.includes(name);
 }
 
 /**
- * Gets all variables available at a specific position in the content
+ * Gets all variables available at a specific position in the content. Includes forloop and tablerowloop variables.
  */
-function getVariablesAtPosition(content: string, position: number): Set<string> {
-  const availableVariables = new Set<string>();
+function getVariablesAtPosition(content: string, position: number): string[] {
+  const availableVariables: string[] = [];
+  const seen = new Set<string>();
 
   // Parse all scopes in the content
   const allScopes = parseAllScopes(content);
 
   // Find scopes that contain the position and collect their variables
   for (const scope of allScopes) {
     if (position >= scope.start && position <= scope.end) {
-      scope.variables.forEach((v) => availableVariables.add(v));
+      scope.variables.forEach((v) => {
+        if (!seen.has(v)) {
+          availableVariables.push(v);
+          seen.add(v);
+        }
+      });
     }
   }
 
-  return availableVariables;
+  return availableVariables.reverse();
+}
+
+export function getVariablesAtPositionWithLoopProperties(content: string, position: number): string[] {
+  const availableVariables = getVariablesAtPosition(content, position);
+  return availableVariables.filter((v) => !v.startsWith('forloop') && !v.startsWith('tablerowloop'));
```

Comment:
we don't want to repeat these variables in the suggestions dropdown, the liquid extension handles it